src/auth/AuthContext.tsx

```diff
@@ -0,0 +1,175 @@
+import React, {
+  createContext,
+  PropsWithChildren,
+  useCallback,
+  useContext,
+  useEffect,
+  useReducer,
+} from 'react';
+import auth, {FirebaseAuthTypes} from '@react-native-firebase/auth';
+import {usePreferences} from '../preferences';
+
+type AuthReducerState = {
+  isInitialized: boolean;
+  abtCustomerId: string | undefined;
+  user: FirebaseAuthTypes.User | null;
+};
+
+type AuthReducerAction =
+  | {
+      type: 'SET_USER';
+      user: FirebaseAuthTypes.User | null;
+    }
+  | {type: 'SET_ABT_CUSTOMER_ID'; abtCustomerId: string | undefined};
+
+type AuthReducer = (
+  prevState: AuthReducerState,
+  action: AuthReducerAction,
+) => AuthReducerState;
+
+const authReducer: AuthReducer = (prevState, action): AuthReducerState => {
+  switch (action.type) {
+    case 'SET_USER': {
+      return {
+        ...prevState,
+        isInitialized: true,
+        user: action.user,
+      };
+    }
+    case 'SET_ABT_CUSTOMER_ID': {
+      return {
+        ...prevState,
+        abtCustomerId: action.abtCustomerId,
+      };
+    }
+  }
+};
+
+const initialReducerState: AuthReducerState = {
+  isInitialized: false,
+  abtCustomerId: undefined,
+  user: null,
+};
+
+type AuthContextState = {
+  signInAnonymously: () => Promise<void>;
+  signOut: () => Promise<void>;
+  updateEmail: (email: string) => Promise<void>;
+  syncAbtCustomer: () => Promise<void>;
+} & AuthReducerState;
+
+const AuthContext = createContext<AuthContextState | undefined>(undefined);
+
+export default function AuthContextProvider({children}: PropsWithChildren<{}>) {
+  const [state, dispatch] = useReducer(authReducer, initialReducerState);
+  const {
+    preferences: {language},
+  } = usePreferences();
+
+  useEffect(() => {
+    if (language) auth().setLanguageCode(language);
```

Comment:
Hm, I'm a bit unsure, seems like an indirect way to get at the language since the language returned from `useTranslation` is changed based on what is the set language in the `PreferencesContext`?

Reply:
My thinking was that we have logic in LanguageContext for determining the app language. Some weeks ago the logic was that Norwegian was default if no preferences set, but it was later changed to that the system language was default if no preferences set. Maybe later it will go back to being Norwegian as default if we include an easy way to switch to English in the onboarding dialog, as have been talked about. By using `language` from `useTranslation` we will piggyback that logic also here, if we want the auth dialog to always be in the same language as the app language.

Reply:
Alright, I can see that argument, preferences does not have logic itself for deciding language. I'll change it 👍🏻